lib/workos/user_management.rb

```diff
@@ -14,6 +14,89 @@ class << self
       extend T::Sig
       include Client
 
+      PROVIDERS = WorkOS::Types::Provider.values.map(&:serialize).freeze
+
+      # Generate an OAuth 2.0 authorization URL that automatically directs a user
+      # to their Identity Provider.
+      #
+      # @param [String] redirect_uri The URI where users are directed
+      #  after completing the authentication step. Must match a
+      #  configured redirect URI on your WorkOS dashboard.
+      # @param [String] client_id This value can be obtained from the API Keys page in the WorkOS dashboard.
+      # @param [String] provider A provider name is used to initiate SSO using an
+      # OAuth-compatible provider. Only 'authkit ,'GoogleOAuth' and 'MicrosoftOAuth' are supported.
+      # @param [String] connection_id The ID for a Connection configured on
+      #  WorkOS.
+      # @param [String] organization_id The organization_id selector is used to
+      # initiate SSO for an Organization.
+      # @param [String] state An arbitrary state object
+      #  that is preserved and available to the client in the response.
+      # @param [String] login_hint Can be used to pre-fill the username/email address
+      # field of the IdP sign-in page for the user, if you know their username ahead of time.
+      # @param [String] domain_hint Can be used to pre-fill the domain field when
+      # initiating authentication with Microsoft OAuth, or with a GoogleSAML connection type.
+      # @example
+      #   WorkOS::UserManagement.authorization_url(
+      #     connection_id: 'conn_123',
+      #     client_id: 'project_01DG5TGK363GRVXP3ZS40WNGEZ',
+      #     redirect_uri: 'https://your-app.com/callback',
+      #     state: {
+      #       next_page: '/docs'
+      #     }.to_s
+      #   )
+      #
+      #   => "https://api.workos.com/user_management/authorize?connection_id=conn_123" \
+      #      "&client_id=project_01DG5TGK363GRVXP3ZS40WNGEZ" \
+      #      "&redirect_uri=https%3A%2F%2Fyour-app.com%2Fcallback&" \
+      #      "response_type=code&state=%7B%3Anext_page%3D%3E%22%2Fdocs%22%7D"
+      #
+      # @return [String]
+      # rubocop:disable Metrics/ParameterLists
+      sig do
+        params(
+          redirect_uri: String,
+          client_id: T.nilable(String),
+          domain_hint: T.nilable(String),
+          login_hint: T.nilable(String),
+          provider: T.nilable(String),
+          connection_id: T.nilable(String),
+          organization_id: T.nilable(String),
+          state: T.nilable(String),
+        ).returns(String)
+      end
+      def authorization_url(
```

Comment:
i see this was copied from the existing `authorization_url` for sso, but all the other SDKs include "get" in the name, like `get_authorization_url`. do you think we should change the Ruby SDK to be consistent with those?

Reply:
I'd opt to be consistent with the other SDKs, I like to append the method on top of it.

Reply:
Updated on https://github.com/workos/workos-ruby/pull/236/commits/8e36fe5d94eaa8bd9933600e23b2485b2b951418

Reply:
@amygdalama Will also update the docs!

Reply:
Generally in Ruby people avoiding naming methods like `get_foo` (and `set_foo`), in favor of `foo` and `foo=` (i.e. `obj.foo = 5`.

So i can see why we called it `authorization_url`. Might be more consistent for _us_, but I don't think consistency across SDK's really benefits the consumer (since they'll likely only use one of them).

Reply:
That's right! Attributes accessing methods, correct?

Yeah, I see your point. I don't have a strong opinion about it, I'd be fine to keep it as `authorization_url`.

Reply:
my b! sounds good to me

Reply:
Method name updated on https://github.com/workos/workos-ruby/pull/236/commits/6356023878e0c407f86fb9b0f5c29c03a9965360